pkg/system/reconciler.go

```diff
@@ -128,6 +129,7 @@ func NewReconciler(
 		AWSCloudCreds:       util.KubeObject(bundle.File_deploy_internal_cloud_creds_aws_cr_yaml).(*cloudcredsv1.CredentialsRequest),
 		AzureCloudCreds:     util.KubeObject(bundle.File_deploy_internal_cloud_creds_azure_cr_yaml).(*cloudcredsv1.CredentialsRequest),
 		GCPCloudCreds:       util.KubeObject(bundle.File_deploy_internal_cloud_creds_gcp_cr_yaml).(*cloudcredsv1.CredentialsRequest),
+		IBMCloudCreds:       util.KubeObject(bundle.File_deploy_internal_cloud_creds_aws_cr_yaml).(*cloudcredsv1.CredentialsRequest),
```

Comment:
I do not see a need for this secret.
Even if you eventually decide to create a secret, it shouldn't be an AWS credential request (cloud credentials operator doesn't support IBM COS).
You can just query and use the secret `ibm-cloud-cos-creds` under `kube-system` directly, also link the BackingStore to it.
Just make sure that it exists and created by the OCP.

Reply:
@jeniawhite What? How can we access kube-system? Why would we?

Reply:
@guymguym 
The cloud credentials operator does not support IBM COS.
The code pulls the credentials from `kube-system` namespace secret `ibm-cloud-cos-creds`.
This secret gets created by OCP when deployed over an IBM cluster environment.
Seems like doing a ref to that secret is allowed (I assume since it looks like it worked in the comments).

Reply:
I see. So it relies on the cluster-permission we have to read and write secrets
here - https://github.com/noobaa/noobaa-operator/blob/681687b5340d16626e4bfd7720b144eb7c99eaf5/deploy/cluster_role.yaml#L24-L31

But the fact that we added this permission was because OBC provisioner did not have any central controller that would be entrusted with the cluster-wide secrets permission. But going forward COSI which will replace OBC will not require us to have cluster-wide secret access, which I perceive as "not our business". 

Isn't OCP going to expose this somehow with cloud-credentials-operator, or any other separate operator that will do that for IBM cloud?

Adding @nimrod-becker

Reply:
Cloud Creds Op won't be minting IBM cloud creds.

There is a talk about moving to a whole different model of security, in which each pod has its own identify and thus can access whatever services it needs. Thought it's not relevant for now, in order for the default to work until the new solution is implemented we need this special handling/

Reply:
Got it. I just added a new comment with a question is if it's possible that this secret be created in our namespace instead of on a kube-system namespace.

Reply:
I think its a chicken and egg thing. The secret needs to be created during deployment of OCP when there is still no NS for OCS ...

Reply:
Our operator won't care if the secret will be created a bit after it started.
It can even be a long time after, it should only wakeup and reconcile.
So at least theoretically it can be imported to the `openshift-storage` namespace only later on when OCS installs/configures.

Reply:
I Don't know if the creds can change or not, if they can just a copy won't be good enough

Reply:
Hi @jeniawhite @guymguym @nimrod-becker 
Thanks for your review comments. 
I have changed the code to look for `ibm-cloud-cos-creds` under the target namespace, and also under the default `openshift-storage` namespace. 
Also updated the code not to use AWS credential request for IBM Cloud.